bundles/framework/announcements/plugin/AnnouncementsPlugin.js

```diff
@@ -0,0 +1,242 @@
+                el = me.templates.main.clone(),
+                header = el.find('div.announcements-header');
+
+            header.append(me._loc.plugin.title);
+            me._bindHeader(header);
+            return el;
+        },
+
+        _bindAnnButton: function (ann, content) {
+            ann.on('click', function () {
+                this.classList.toggle('active');
+                if (content[0].style.maxHeight) {
+                    content[0].style.visibility = 'hidden';
+                    content[0].style.padding = null;
+                    content[0].style.maxHeight = null;
+                } else {
+                    content[0].style.visibility = 'visible';
+                    content[0].style.padding = '5px';
+                    content[0].style.maxHeight = content[0].scrollHeight + 'px';
+                }
+            });
+        },
+
+        _bindHeader: function (header) {
+            var me = this;
+            header.on('click', function () {
+                if (me.open) {
+                    me.closeSelection();
+                } else {
+                    me.openSelection();
+                }
+            });
+        },
+
+        /**
+        * @method addAnnouncements
+        * Adds announcements to the list from config
+        */
+        addAnnouncements: function () {
+            var me = this;
+            var announcementsIds = me._config.announcements;
+
+            var announcements = [];
+
+            for (const i of announcementsIds) {
+                for (const j of me.allAnnouncements) {
+                    if (j.id === i) {
+                        announcements.push(j);
+                    }
+                }
+            }
+
+            if (!me.open) {
+                delete me.announcementsContent;
+            } else {
+                jQuery('div.announcements-content').children().remove();
+            }
+            delete me.annRefs;
+            me.annRefs = {};
+
+            announcements.forEach(announcement => {
+                if (me.annRefs[announcement.id]) {
+                    // already added
+                    return;
+                }
+
+                if (!me.announcementsContent) {
+                    me.announcementsContent = me.templates.announcementsContent.clone();
+                    if (me.open) {
+                        jQuery('div.mapplugin.announcements').append(me.announcementsContent);
+                    }
+                }
+
+                var announcementsDiv = me.announcementsContent.find('div.announcements-content'),
+                    div = me.templates.announcement.clone();
+
+                const annTime = announcement.begin_date.replace(/-/g, '/') + ' - ' + announcement.end_date.replace(/-/g, '/');
+
+                div.find('button').append(announcement.title);
+                div.find('div#announcement-description').append(announcement.content);
+                div.find('div#announcement-time').append(annTime);
```

Comment:
This will generate errors for conflicting ids on the dev console when there are more than one announcement since you repeat the same id for all. Use classes instead.